src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb

```diff
@@ -738,6 +662,162 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
             Return binder
         End Function
 
+        Private Shared Function AddImportForRecord(
+            importRecord As ImportRecord,
+            importBinder As Binder,
+            metadataDecoder As MetadataDecoder,
+            position As Integer,
+            ByRef importsBuilder As ArrayBuilder(Of NamespaceOrTypeAndImportsClausePosition),
+            ByRef aliases As Dictionary(Of String, AliasAndImportsClausePosition),
+            ByRef xmlImports As Dictionary(Of String, XmlNamespaceAndImportsClausePosition)) As Boolean
+
+            Dim targetString = importRecord.TargetString
+
+            ' NB: It appears that imports of generic types are not included in the PDB, so we never have to worry about parsing them.
+            ' NB: Unlike in C# PDBs, the assembly name will not be present, so we have to just bind the string.
+            Dim targetSyntax As NameSyntax = Nothing
+            If Not String.IsNullOrEmpty(targetString) AndAlso ' CurrentNamespace may be an empty string, new-format types may be null.
+                    importRecord.TargetKind <> ImportTargetKind.XmlNamespace AndAlso
+                    Not TryParseDottedName(targetString, targetSyntax) Then
+
+                Debug.WriteLine($"Import record '{importRecord}' has syntactically invalid target '{targetString}'")
+                Return False
+            End If
+
+            ' Check for syntactically invalid aliases.
+            Dim [alias] = importRecord.Alias
+            If Not String.IsNullOrEmpty([alias]) Then
+                Dim aliasNameSyntax As NameSyntax = Nothing
+                If Not TryParseDottedName([alias], aliasNameSyntax) OrElse aliasNameSyntax.Kind <> SyntaxKind.IdentifierName Then
+                    Debug.WriteLine($"Import record '{importRecord}' has syntactically invalid alias '{[alias]}'")
+                    Return False
+                End If
+            End If
+
+            Select Case importRecord.TargetKind
+                Case ImportTargetKind.Type
+                    Dim typeSymbol As TypeSymbol
+                    Dim portableImportRecord = TryCast(importRecord, PortableImportRecord)
+                    If portableImportRecord IsNot Nothing Then
+                        typeSymbol = portableImportRecord.GetTargetType(metadataDecoder)
+
+                        If typeSymbol Is Nothing Then
+                            Debug.WriteLine($"Import record '{importRecord}' requires unknown type")
+                            Return False
```

Comment:
I think you're right.  I think I refactored an if incorrectly.  I've changed it to an assert to match C#.